### ISSUE TITLE
Upgrade icons to modern SVGs

### ISSUE THREAD

**Issue by tonihele** [jMonkeyEngine/sdk#355]
Upgrade icons to modern SVGs. SVGs are supported ever since Netbeans 11.x. With all the HiDPI stuff and all that, I propose a little bit of a facelift. Take SVGs into use as icons for scaling.  Fortunately we don't have a lot of these.

Icons are here with the constants that need to be changed as well:
https://github.com/jMonkeyEngine/sdk/tree/master/jme3-core/src/com/jme3/gde/core/icons

At least I don't posses the capabilities to create new icons but there should be plenty available in the Internets. Just need to be careful with the licenses.

**Comment by neph1** [comment]
Good idea. I've been using icons from here, lately: https://game-icons.net/tags/gui.html
May not have all needed, but can be combined in inkscape, for example. Liberal license.
If I get the time I might dig into this.

**Comment by neph1** [comment]
More icons here:
jme3-core/src/com/jme3/gde/core/editor/icons
For the shader node editor, it seems.

**Comment by tonihele** [comment]
> I did a naive attempt at replacing a couple of icons, but just loading an svg didn't work. It's either not shown, or extremely small, like a pixel.

Could it be that the format of the SVG is not in the form expected? Needs to have width and height specified it seems.

https://github.com/apache/netbeans/pull/1278/files#diff-9ba09915eb7eaa1837fdd0943744d49bc050900b2a809f09db8fe6c320b81f69R32

**Comment by neph1** [comment]
This is what it looks like:
`<svg style="height: 512px; width: 512px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">`
I also tried various heights/widths and changed the viewBox, as well as scale, but they didn't seem to change anything.

**Comment by neph1** [comment]
Hmm, I had checked that PR, but didn't think of looking for supplied svg's. They do look different. I'll see if I can wrangle the ones I have to fit that format.

**Comment by neph1** [comment]
Bingo, I guess:
![image](https://user-images.githubusercontent.com/7988802/183471557-a5e7a037-99d6-494d-85d0-7e6745ca8d42.png)
I had to be specific about the width/height being 16px for it to fit.

I've been bit conservative and selected icons that look like the existing ones, but it would be a good opportunity for someone with UI inclinations to help pick new/better icons. I'm pretty sure (and glad) we won't find anything looking like the chimp icons.

**Comment by tonihele** [comment]
Heh I did find actually very cool chimp icons that would have fit the current style perfectly. Although, one needs to pay 20e or so from them but after that they are free to use without restrictions. I could acquire them if people want. But sadly no brains blown out chimp there :D

If I understood correctly, the benefit from using SVGs comes with those HiDPI things. Then the UI (and its icons) are scaled up and pixels start to show. So yeah, little bit niche thing but but but... maybe think of it as a general facelift also. The chimps are quite pixellated in my opinion by default now.

**Comment by neph1** [comment]
When thinking about this one step further: "Wouldn't it be cool if" textures had an icon that was actually the texture, scenes had an icon that was a screenshot of the scene?
What would be required for that?

- Screenshot of scene when saving using Robot (or is that legacy?). Textures are images, so that should be easy. Materials also have image representations.
- Saving images to an external folder (.local, etc).
- Reading from said folder.
- Custom icon per asset, fallback to asset type icon.

Maybe pointless as long as the icons are only 16x16.

**Comment by neph1** [comment]
More(!) icons here: https://github.com/jMonkeyEngine/sdk/tree/master/jme3-scenecomposer/src/com/jme3/gde/scenecomposer

**Comment by tonihele** [comment]
Here is by the way how they do (did.. I'm not sure is this super accurate) this in NB itself:
![image](https://user-images.githubusercontent.com/8344766/200171912-9e01d7c1-1753-405f-9a9f-2c735307fbb3.png)

Notice that the outline is a darker shade of the actual item. This probably helps them to be seen in various themes. And this would probably help the green on blue icons we have:
![image](https://user-images.githubusercontent.com/8344766/200172118-ce660656-fd80-4e6e-944a-615eb59260d2.png)

Just a thought.

**Comment by neph1** [comment]
Nice thought ;) I had not considered it before doing the icons, and when I added the outline I scripted it (for most of them). Maybe it's just a matter of opacity, in which case they can be "sed'ed" again. But next version?

**Comment by tonihele** [comment]
3.5 is already out :) By all means this can be tried out. I didn't try it at all, so I'm unsure what results it will yield.

**Comment by neph1** [comment]
Sorry, I missed the fanfares :D

Well, you made me have to check. The stroke paint is half over the paint and half outside, so opacity doesn't give exactly that result.
Here it is with opacity and a bit of blur to give it some pleasant smoothness.
![image](https://user-images.githubusercontent.com/7988802/200174389-e8028588-0893-4e68-9ebf-d63c9872bed6.png)

**Comment by tonihele** [comment]
> Sorry, I missed the fanfares :D
> 
> Well, you made me have to check. The stroke paint is half over the paint and half outside, so opacity doesn't give exactly that result. Here it is with opacity and a bit of blur to give it some pleasant smoothness. ![image](https://user-images.githubusercontent.com/7988802/200174389-e8028588-0893-4e68-9ebf-d63c9872bed6.png)

Looks ok like that. The challenge comes with the small size and the background color. Can't say how they perform. I put all my trust in you :)

**Comment by neph1** [comment]
I'll experiment a bit in the upcoming PR's, and if we find something that works well, I'll go back and update the existing ones.
